backend/app/categories/controller.py

```diff
@@ -38,8 +39,17 @@ async def get_category_route(category_id: CategoryEnum,
 async def get_category_item_route(category_id: CategoryEnum, item_id: ObjectID,
                                   db: AsyncIOMotorClient = Depends(get_database)) -> ItemInResponse:
     """Get the details about a particular item"""
-    # TODO: add particular item fetching
-    return {"success": True}
+
+    if (ObjectId.is_valid(item_id)==False):
+        """ Validation for item_id """
+        raise HTTPException(
+            status_code=400, detail='Not a valid ObjectID')
+    try:
+        _res = await db[category_id]["data"].find_one({"_id": ObjectId(f"{item_id}")}) 
+        return ItemInResponse(data=_res)    
+    except:
+        raise HTTPException(
+            status_code=400, detail=f'ObjectID {item_id} not found in {category_id}')
```

Comment:
Instead of `try/except` it's better to just do something like
```python
item = await db[category_id]["data"].find_one({"_id": item_id})
if item:
    return ItemInResponse(data=item)
raise HTTPException(status_code=404, detail=f"item not found in {category_id}")
```

Reply:
Is this the general convention for doing stuff like this?
Or is it like (performance?) improvements?

Reply:
General convention